### ISSUE TITLE
Refactor configuration 

### ISSUE THREAD

**Issue by markets** [sidekiq-cron/sidekiq-cron#491]
We should consolidate all configuration options into `Sidekiq::Cron.configure` block, and leave the `Sidekiq::Options` module to access Sidekiq settings (read-only, seems a better idea to avoid changing Sidekiq settings). To avoid breaking changes, maybe we can make the new flags to fallback to `Sidekiq::Options`.

**SUMMARY**
- `Sidekiq::Cron.configure` => to configure Sidekiq-Cron
- `Sidekiq::Options` => module to read Sidekiq settings

**Comment by markets** [comment]
@satyakampandya Would you like to take a look to this one? I want to launch `rc2` after this refactor.

**Comment by satyakampandya** [comment]
@markets I'll have a look at this tonight.

**Comment by satyakampandya** [comment]
@markets Would you mind taking a look at this commit? https://github.com/satyakampandya/sidekiq-cron/commit/cd0270bc16ee78f388cdbaa6d379bc80216deb61

Just want to confirm if I'm heading in the right direction with this refactor.

**Comment by markets** [comment]
Exactly! This is the direction I had in mind 👍

**Comment by satyakampandya** [comment]
@markets I encountered an issue while migrating `cron_poll_interval` from `Sidekiq::Options` to `Sidekiq::Cron.configure`. Specifically, I'm seeing the following error:

```
NoMethodError: undefined method `capsules` for #<Sidekiq::Cron::Configuration:0x000000011b36b368>
```
It appears that capsules is necessary for Sidekiq::Options, but it doesn't apply to Sidekiq::Cron.configuration. We don't need this feature in the `Sidekiq::Cron.configure` context. Could you advise on how we should handle this difference?

You can track progress [here](https://github.com/satyakampandya/sidekiq-cron/tree/refactor-sidekiq-cron-configurations).

**Comment by markets** [comment]
@satyakampandya You'll need to check the Launcher and Poller classes to understand how we inject this to Sidekiq config. This flag, in fact, has some extra work... 😅 But anyway, the idea is to keep the public api under `Sidekiq::Cron.configure`, which is more clear for our users.

PS just checked your progress and it looks really good 👌

**Comment by markets** [comment]
Closed by #495
